Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/CopyrightPresentAnalyzer.cs

```diff
@@ -29,60 +28,41 @@ public CopyrightPresentAnalyzer(Helper helper)
 					@"Copyright Present",
 					@"File should start with a copyright statement, containing the company name, the year and either © or 'Copyright'.",
 					@"File should start with a comment containing the company name, the year and either © or 'Copyright'.",
-					Categories.Documentation
-				 )
-		{
-			_helper = helper;
-		}
-
-
-		public override void Initialize(AnalysisContext context)
-		{
-			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
-			context.EnableConcurrentExecution();
-			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.CompilationUnit);
-		}
+					Categories.Documentation, helper)
+		{ }
 
-		private void Analyze(SyntaxNodeAnalysisContext context)
+		protected override void Analyze()
 		{
-			CompilationUnitSyntax node = (CompilationUnitSyntax)context.Node;
-
-			GeneratedCodeDetector generatedCodeDetector = new();
-			if (generatedCodeDetector.IsGeneratedCode(context))
+			if (Helper.IsAssemblyInfo(Context) || Helper.HasAutoGeneratedComment(Node))
```

Comment:
This could be moved to the base class. Same holds of `IsInTestClass`

Reply:
Agreed. We could keep poking at the Helper class over time.